docs/udf.rst

```diff
@@ -0,0 +1,179 @@
+.. _ksql-udfs:
+
+KSQL User Defined Function Reference
+====================================
+
+KSQL supports creating User Defined Functions (UDFs) via custom jars that are
+uploaded to the ext directory of the KSQL installation. At start up time KSQL scans the
+jars in the directory looking for any classes that annotated with ``@UdfDescription``. It then
+searches the class for any public methods that are annotated with ``@Udf``. Each UDF that is found
+is parsed and, if successful, loaded into KSQL.
+
+Each UDF instance has its own child-first ``ClassLoader`` that is isolated from other UDFs. If you
+need to use any third-party libraries with your UDFs then they should also be part of your jar, i.e.,
+you should create an "uber-jar". The classes in your uber-jar will be loaded in preference to any
+on the KSQL classpath excluding anything that is vital to the running of KSQL, i.e., classes that are
+part of ``org.apache.kafka`` and ``io.confluent``. Further, the ``ClassLoader`` can restrict access
+to other classes via a blacklist. The blacklist file is ``resource-blacklist.txt``. You can add
```

Comment:
Where would have one have to store the blacklist file?  Inside the custom jar?  If so, where exactly in there, inside the root directory?

Reply:
is further down in the section for blacklisting